### PR TITLE
Use a temp checkout path for automation when building docs

### LINKED ISSUE #2900: GH Pages deployments failing
## Description

Consider these two workflow runs

- https://github.com/WordPress/openverse/actions/runs/5971194440/job/16209593998#step:2:566
- https://github.com/WordPress/openverse/actions/runs/5978165187/job/16219691937#step:2:564

Both fail with the following error message:

```
/usr/bin/git submodule sync --recursive
/usr/bin/git -c protocol.version=2 submodule update --init --force --depth=1 --recursive
Error: fatal: No url found for submodule path 'automation-checkout' in .gitmodules
Error: The process '/usr/bin/git' failed with exit code 128
```

This error has caused a number of GitHub Pages deployments to fail recently, with our documentation and PR previews not updating as expected.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #2900 by @dhruvkb

## Description

Fixes an issue with #2647 which was causing the documentation publishing step to fail because a repo was being checked out inside of another repo (namely the automation used during the CI steps was being checked out to `<workspace>/automation-checkout` when the actual `gh-pages` branch was being checked out to `<workspace>` itself).

## Testing Instructions

~~I'm not sure the best way to test this, I'll follow up with a small doc change which should at least trigger the preview, but we may need to wait for a legitimate docs change in order to see if it is successful or not.~~

I've added a legitimate documentation change, so hopefully we'll be able to see the results once this is merged.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
Good catch! I spent several hours but could not figure out from where Git got the idea that we have submodules.

**Comment by AetherUnbound** [comment]
> Good catch! I spent several hours but could not figure out from where Git got the idea that we have submodules.

I did a lot of random searching in our repo and ran the _exact_ steps the CI was running to come across this 😅 I'm disappointed we didn't catch this earlier honestly, since the deployments were pulling down the `gh-pages` branch and deploying to that 🤔 I hope this does the trick!

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
The action fails on `tmp` not being in the working directory.

To fix this, it might be necessary to move both `gh-pages` _and_ the automations checkout into separate paths:

```diff
diff --git a/.github/workflows/ci_cd.yml b/.github/workflows/ci_cd.yml
index ea45cbfbd..82f5850e4 100644
--- a/.github/workflows/ci_cd.yml
+++ b/.github/workflows/ci_cd.yml
@@ -880,6 +880,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           ref: gh-pages
+          path: gh-pages
 
       - name: Checkout automations from repository
         uses: actions/checkout@v3
@@ -890,7 +891,7 @@ jobs:
         if: github.event_name == 'push'
         run: |
           mv /tmp/docs /tmp/gh-pages
-          mv _preview /tmp/gh-pages/_preview
+          mv gh-pages/_preview /tmp/gh-pages/_preview
 
       - name: Replace preview of current PR
         if: github.event_name == 'pull_request'
```

**Comment by sarayourfriend** [comment]
Also to re-iterate what Dhruv said, I was stumped on why it thought we had submodules. Good find.

**Comment by obulat** [comment]
I tried re-running the CI, and it failed again. It seems to have failed because "Copy existing previews" step did not run, so `/tmp/gh-pages/_preview/` probably doesn't exist. This means that `mv /tmp/docs /tmp/gh-pages/_preview/${{ github.event.pull_request.number }}` fails because `/tmp/gh-pages/_preview/` doesn't exist.

Should the `if` for "Copy existing previews" step be updated to something other than `github.event_name == 'push'`?

Steps that cause CI failure:
```
      - name: Copy existing previews
        if: github.event_name == 'push'
        run: |
          mv /tmp/docs /tmp/gh-pages
          mv gh-pages/_preview /tmp/gh-pages/_preview

      - name: Replace preview of current PR
        if: github.event_name == 'pull_request'
        run: |
          cp -r . /tmp/gh-pages
          sudo rm -rf /tmp/gh-pages/_preview/${{ github.event.pull_request.number }}
          mv /tmp/docs /tmp/gh-pages/_preview/${{ github.event.pull_request.number }}
```

**Comment by obulat** [comment]
> should be

I added this line, and the "emit docs" step worked. Unfortunately, the preview link still doesn't work :(

File: .github/workflows/ci_cd.yml [hunk 1]
```diff
@@ -824,12 +825,12 @@ jobs:
         if: github.event_name == 'push'
```
Comment (obulat) [inline .github/workflows/ci_cd.yml:825]:
```suggestion
```

**Comment by AetherUnbound** [comment]
> I added this line, and the "emit docs" step worked. Unfortunately, the preview link still doesn't work :(

It looks like it has for me! Do you mind checking again @obulat?

**Pull request merged**